core/trino-main/src/main/java/io/trino/memory/NodeMemoryConfig.java

```diff
@@ -104,4 +106,18 @@ public NodeMemoryConfig setHeapHeadroom(DataSize heapHeadroom)
         this.heapHeadroom = heapHeadroom;
         return this;
     }
+
+    @NotNull
+    public DataSize getMaxTaskDescriptorStorageMemory()
```

Comment:
`NodeMemoryConfig` is more about workers, right? Would `MemoryManagerConfig` be a better place for this one?

Reply:
It looks like `MemoryManagerConfig` is specific to `ClusterMemoryManager`. But I agree, `NodeMemoryConfig` feels to be more specific to workers. I'm not sure if this config is worth a separate configuration file. What do you think about moving this config to the `QueryManagerConfig`? Currently this class contains many configuration parameters related to query / split scheduling.

Reply:
Fine for me

Reply:
Discussed offline. Moved to `QueryManagerConfig`.